test/unit/gossip-client-daemonset.bats

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env bats
+
+load _helpers
+
+@test "global/gossipEncryption: disabled in client DaemonSet by default" {
+  cd `chart_dir`
+  local actual=$(helm template \
+      -x templates/client-daemonset.yaml  \
+      . | tee /dev/stderr |
+      yq '.spec.template.spec.containers[] | select(.name=="consul") | .env[] | select(.name == "GOSSIP_KEY") | length > 0' | tee /dev/stderr)
+  [ "${actual}" = "" ]
+}
+
+@test "global/gossipEncryption: disabled in client DaemonSet when servers are disabled" {
+  cd `chart_dir`
+  local actual=$(helm template \
+      -x templates/enterprise-license.yaml  \
```

Comment:
This should presumably still be templating the client daemonset. Additionally, I would expect that client gossip encryption could be enabled even if servers within the Kubernetes cluster are disabled. If the clients are talking to an external Consul cluster, they still might need a gossip encryption key, right?

Oh, or maybe this was a copy/paste issue and you meant the clients were disabled?

```suggestion
      -x templates/client-daemonset.yaml  \
```